src/serialization.hpp

```diff
@@ -43,29 +44,47 @@ class OutputGetter {
     T outputSource;
 };
 
+template <typename ProtoStorage, typename ProtoType>
+class ProtoGetter {
+    ProtoStorage protoStorage;
+
+public:
+    ProtoGetter(ProtoStorage protoStorage) :
```

Comment:
Wouldn't it be passed by copy?

Reply:
It depends on what type you create -
if
ProtoGetter<tensorflow::serving::PredictResponse*, tensorflow::TensorProto&> then no
if
ProtoGetter<tensorflow::serving::PredictResponse, tensorflow::TensorProto> then yes